suite/suite.go

```diff
@@ -63,6 +63,20 @@ func failOnPanic(t *testing.T) {
 	}
 }
 
+// Run provides suite functionality around golang subtests.  It should be
```

Comment:
I think we need to add some additional documentation concerning the potential impact of calling [T.Parallel](https://golang.org/pkg/testing/#T.Parallel), especially as it can impact subsequent calls to functions dependent on the state of `Suite`, such as `Suite.Assert`, `Suite.Require`, and `Suite.T`. If someone calls `suite.T().Parallel()` within the subtest, then the `T.Run` call within `Suite.Run` will return and so will `Suite.Run`. This could lead to another test or subtest running that modifies the suite's testing context, modifying state for the still running first now parallel subtest. Users will need to be careful regarding when they call `T.Parallel()` and that they already have any testing context (`Suite.T()`), assertion context (`Suite.Assert()`), or require context (`Suite.Require()`) pulled out that they might need prior to calling `T.Parallel`